dig.go

```diff
@@ -189,6 +197,8 @@ type provider interface {
 	// The values produced by this provider should be submitted into the
 	// containerStore.
 	Call(containerStore) error
+
+	ID() uintptr
```

Comment:
nit: I would make this the first one in the list since it's the unique identifier.

(Also, consider adding a comment above this.)

    // ID is a unique numerical identifier for this provider.